### PR TITLE
Add support to specify different annotation for the task and web deployments in AWX operator

### LINKED ISSUE #1331: Add support to specify different annotation for the task and web deployments in AWX operator 2.x.x
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that AWX Operator is open source software provided for free and that I might not receive a timely response.

### Feature Summary

To specify the annotations, we use the `annotations` keyword in the AWX manifest:
```
---
apiVersion: "awx.ansible.com/v1beta1"
kind: "AWX"
metadata:
  name: "awx"
spec:
  annotations: |
    foo: "bar"
```
This works fine in pre-2 releases. In 2.0.0, the AWX operator creates two deployments: task and web. If annotations are the same, everything is fine, but if we want to specify different annotations, we have a problem.

Here is our use case. We use HashiCorp Vault to specify DB (and other) secrets for AWX. The HC Vault configuration is specified using annotations. We are using a self-signed cert inside the cluster, so we need a CA cert in the Vault side container. To do that, we create a ConfigMap that includes the CA cert. We mount the CA cert to the `awx-web` container. Then we use the annotation:
```
vault.hashicorp.com/agent-copy-volume-mounts: "awx-web"
```
This annotation tells Vault to copy volume mounts from the `awx-web` container to the Vault side container, which is then used for to verify the HC Vault cert.

Because in 2.0.0, there is no `awx-web` container in the "task" deployment, the task containers crash.

We suggest having the following options:
* `annotations`: Common annotations for both deployments
* `task_annotations`: Annotations specific to the task deployment
* `web_annotations`: Annotations specific to the web deployment

### PR DESCRIPTION
**Pull request by djyasin**
##### SUMMARY

Fixes #1331

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### Test-Case Scenarios
- 4 Test cases total
    - No annotations
        - Web and task should deploy successfully with no additional annotations
    - Just `web_annotations`
        - Only web deployment has custom annotations, matches what was set
           - Added example from defaults to spec, checked web deployment to ensure they were present, then checked other deployments to be sure they were not there. 
    - Just `task_annotations`
       - Only task deployment has  custom annotations, matches what was set
    - Common `annotations` for both deployments
       - Both deployments have common annotations 
    - Override scenario, have `web_annotations`, `task _annotations`, and `annotations`
       - Ensure these remain distinct

### CODE REVIEW

File: roles/installer/templates/deployments/task.yaml.j2 [hunk 1]
```diff
@@ -44,7 +44,9 @@ spec:
         checksum-secret-{{ secret }}: "{{ lookup('ansible.builtin.vars', secret, default='')["resources"][0]["data"] | default('') | sha1 }}"
 {% endfor %}
 {% if annotations %}
-        {{ annotations | indent(width=8) }}
+  annotations: {{ annotations | indent(width=8) }}
+{% elif task_annotations %}
+  task_annotations: {{ task_annotations | indent(width=8) }}
```
Comment (TheRealHaoLiu) [inline roles/installer/templates/deployments/task.yaml.j2]:
```suggestion
  annotations: {{ task_annotations | indent(width=8) }}
```

File: roles/installer/templates/deployments/web.yaml.j2 [hunk 2]
```diff
@@ -45,7 +45,9 @@ spec:
         checksum-secret-{{ secret }}: "{{ lookup('ansible.builtin.vars', secret, default='')["resources"][0]["data"] | default('') | md5 }}"
 {% endfor %}
 {% if annotations %}
-        {{ annotations | indent(width=8) }}
+  annotations: {{ annotations | indent(width=8) }}
+{% elif web_annotations %}
+  web_annotations: {{ web_annotations| indent(width=8) }}
```
Comment (TheRealHaoLiu) [inline roles/installer/templates/deployments/web.yaml.j2]:
```suggestion
  annotations: {{ web_annotations | indent(width=8) }}
```

File: config/crd/bases/awx.ansible.com_awxs.yaml [hunk 3]
```diff
@@ -183,7 +183,13 @@ spec:
                 description: Additional labels to apply to the service
                 type: string
               annotations:
-                description: annotations for the pods
+                description: Common annotations for both deployments
+                type: string
+              task_annotations:
+                description: Annotations specific to the task deployment
```
Comment (TheRealHaoLiu) [inline config/crd/bases/awx.ansible.com_awxs.yaml]: is this an override of `annotations` field or is it a combination with the `annotation`

Comment (TheRealHaoLiu) [reply config/crd/bases/awx.ansible.com_awxs.yaml]: from the code this looks like a override but the description here sounds like it will be additive rather than an override

**Comment by TheRealHaoLiu** [comment]
please update x-descriptor for the ClusterServiceVersion resource 

example
https://github.com/ansible/awx-operator/blob/0c2f45569cadd3d41be4375a91927623e60e1b14/config/manifests/bases/awx-operator.clusterserviceversion.yaml#L688-L692

File: config/crd/bases/awx.ansible.com_awxs.yaml (hunk 3 above)
Comment (rooftopcellist) [reply config/crd/bases/awx.ansible.com_awxs.yaml]: Maybe rewording it to say:
```
                description: Task deployment annotations. This will override the general annotations parameter for the Task deployment.
```

File: roles/installer/defaults/main.yml [hunk 4]
```diff
@@ -184,6 +184,18 @@ web_affinity: {}
 #   my.annotation/2: value2
 annotations: ''
 
+# Add annotations to awx task pods. Specify as literal block. E.g.:
```
Comment (TheRealHaoLiu) [inline roles/installer/defaults/main.yml]: s/Add/Override/

**Review by jjwatt — APPROVED** [review]
Paired with @djyasin @TheRealHaoLiu @gamuniz coding, debugging and testing. LGTM 🤌

**Review by gamuniz — APPROVED** [review]
